packages/material-ui/src/Button/Button.js

```diff
@@ -409,7 +409,7 @@ Button.propTypes = {
   /**
    * The variant to use.
    */
-  variant: PropTypes.oneOf(['contained', 'outlined', 'text']),
+  variant: PropTypes.oneOfType([PropTypes.oneOf(['contained', 'outlined', 'text']), PropTypes.string]),
```

Comment:
This was manually added, the generated prop type is `string` 😢

Reply:
Is this still an issue, maybe @eps1lon could help, at least we have this `/* @typescript-to-proptypes-ignore */` option

Reply:
Because technically `'foo' | string` will accept any `string`. There are some hacks to get at least IntelliSense working but we should be aware that this removes any typo(sic!)-safety from `variant`.

I haven't looked at the full API but if we could use module augmentation then we could keep typo-safety for standard variants.